applications/endoscopy_tool_tracking_distributed/python/video_input_fragment.py

```diff
@@ -0,0 +1,125 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from holoscan.core import Fragment
+from holoscan.operators import AJASourceOp, VideoStreamReplayerOp
+
+# Enable this line for Yuam capture card
+# from holohub.qcap_source import QCAPSourceOp
+
+
+class VideoInputFragment(Fragment):
+    @property
+    def is_overlay_enabled(self):
+        return self._is_overlay_enabled
+
+    @is_overlay_enabled.setter
+    def is_overlay_enabled(self, value):
+        self._is_overlay_enabled = value
+
+    @property
+    def source_block_size(self):
+        print("Getting value")
+        return self._source_block_size
+
+    @source_block_size.setter
+    def source_block_size(self, value):
+        self._source_block_size = value
+
+    @property
+    def source_num_blocks(self):
+        return self._source_num_blocks
+
+    @source_num_blocks.setter
+    def source_num_blocks(self, value):
+        self._source_num_blocks = value
+
+    @property
+    def width(self):
+        return self._width
+
+    @width.setter
+    def width(self, value):
+        self._width = value
+
+    @property
+    def height(self):
+        return self._height
+
+    @height.setter
+    def height(self, value):
+        self._height = value
+
+    @property
+    def rdma(self):
+        return self._rdma
+
+    @rdma.setter
+    def rdma(self, value):
+        self._rdma = value
+
+    def __init__(self, app, name, source, video_dir):
```

Comment:
Usually I would extend the `__init__` method with additional kwargs such as `rdma`, `width` and `height` instead of using setters as above but I think that is also okay.

Reply:
I removed the setters since they were needed.